### ISSUE TITLE
[Question] Re-use WebSocketApp connection and share between onMessage and recv()

### ISSUE THREAD

**Issue by lswest** [websocket-client/websocket-client#562]
Hi,

I've got a bit of a difficult use-case. I'm working on a multithreaded program that needs to utilize a websocket connection to initiate a refresh of settings (stored in module-wide variables). However, I also need to use the websocket connection to request specific information from within other threads. I thought I could solve this by storing the WebSocketApp connection in the global variables, and then trigger/receive the messages with `settings.ws.send()` and `settings.ws.sock.recv()`. However, this works exactly once... Any subsequent attempts result in the thread that called the` .recv()` function to wait endlessly, while the response is handled in the `onMessage` function.

My best guess is that the `settings.ws.sock.recv()` responds once because the dispatcher is not yet active, but on subsequent calls directs the responses to `onMessage()`.

So, to sum it up, here are the questions:

1. Is it possible to manually receive a response outside of on_message when running a `websocket.WebSocketApp`?
2. Would this approach be possible if I wrote my own class or stored the actual create_connection in a variable? I don't know enough about websockets to judge whether what I'm doing is actually possible/intended with a websocket connection.

Thanks in advance!

**Comment by lucashussey** [comment]
To make sure I understand, are you trying to use a websocket to send messages between internal (within the client) threads? If so, you may want to use Queue, which you can pass as an argument when you start each thread.

In general, your problem might be a locking issue, where only 1 thread can access a shared object at any given time.

**Comment by engn33r** [comment]
Closing due to inactivity, but I can reopen if this issue still exists and more information is provided.

I know this is an old issue, but if it is still a problem, please share a bit more about your scenario, ideally providing the information outlined in the [contributing guidelines](https://github.com/websocket-client/websocket-client/blob/master/CONTRIBUTING.md). I am optimistic the recent updates to the [project's threading documentation](https://websocket-client.readthedocs.io/en/latest/threading.html) might help clarify how websocket-client supports multiple threads, but seeing some sample code of what you have tried would be most helpful for me.
